src/mongo/db/curop.h

```diff
@@ -305,6 +306,10 @@ class CurOp {
         if (_dbprofile <= 0)
             return false;
 
+        if (serverGlobalParams.rateLimit > 1 && _dbprofile >= 2 && ms < serverGlobalParams.slowMS) {
+            return _prng.nextCanonicalDouble() * serverGlobalParams.rateLimit < 1.0;
```

Comment:
You're right, the snippet is wrong.
Is `_prng.nextInt64(serverGlobalParams.rateLimit) >= 0` better?
In any case, you are able find the proper equation by yourself :)